src/module-elasticsuite-core/Helper/IndexSettings.php

```diff
@@ -93,41 +143,47 @@ public function getLanguageCode($store)
     }
 
     /**
-     * Create a new index for an identifier (eg. catalog_product) by store including current date.
+     * Create a new index for an identifier (e.g. catalog_product) by store including current date.
      *
-     * @param string                                                $indexIdentifier Index identifier.
-     * @param integer|string|\Magento\Store\Api\Data\StoreInterface $store           The store.
+     * @param string                        $indexIdentifier Index identifier.
+     * @param integer|string|StoreInterface $store           The store.
      *
      * @return string
      */
     public function createIndexNameFromIdentifier($indexIdentifier, $store): string
     {
-        $indexNameSuffix = $this->getIndexNameSuffix(new Zend_Date());
+        $indexNameSuffix = $this->getIndexNameSuffix(new DateTime());
 
         return sprintf('%s_%s', $this->getIndexAliasFromIdentifier($indexIdentifier, $store), $indexNameSuffix);
     }
 
     /**
      * Get index name suffix.
      *
-     * @param Zend_Date $date Date
+     * @param DateTime $date Date
      * @return string
      */
-    public function getIndexNameSuffix(Zend_Date $date): string
+    public function getIndexNameSuffix(DateTime $date): string
```

Comment:
Can you try to add a unit test to this method as well ? 

The test should mock the fact that the pattern can be the old one or the new one.